### ISSUE TITLE
Delauney mesh of scan grids

### ISSUE THREAD

**Issue by tpwrules** [CloudCompare/CloudCompare#2077]
**Describe the feature you would like**
I would like the ability to create a Delauney 2.5D mesh of a TLS scan grid, as opposed to just meshing adjacent grid points which is performed by the existing option.

**Describe alternatives you've considered**
Not sure what else might be useful, but this seems relatively simple to solve the problem.

**Is your feature request related to a problem? Please describe.**
It's not easy to generate a good mesh of a large scan grid. Just meshing produces a mesh with hundreds of millions of triangles which is slow to work with. Subsampling then meshing using a scan grid produces very few triangles (as triangles are not created if adjacent grid points are not present). Subsampling then meshing using Poisson reconstruction is subject to the problems of that method. Meshing then subsampling is not possible as CloudCompare cannot subsample meshes.

**Additional context**

**Comment by dgirardeau** [comment]
Sorry, I'm not sure to understand what you mean exactly by "Delauney mesh of scan grids" if it's not meshing regularly with the scan grid... Can you give us more details?

**Comment by tpwrules** [comment]
Basically the current “mesh scan grid” function only works if all the points are still in the cloud. I want it to be possible to use that function or a similar one after, say, a random 1% subsampling operation.

I think this is possible by using the Delaunay meshing algorithm in “grid space” instead of the 3D point cloud if that makes sense.

**Comment by RyanJMcGowan** [comment]
That doesn't sound like a Delaunay mesh. Do you mean creating a digital elevation model (DEM) which is a grid representation of the z values of a surface? If so, these are often stored as TIFFs and can be made by rasterizing the z values with pixel spacing at the desired sampling rate. This also effectively subsamples the cloud data to a specific distance.

If you need a mesh of a DEM, you can export the mesh after rasterizing.
![image](https://github.com/user-attachments/assets/ddedb12a-61cd-4d49-aae3-6ff9e5b144dc)

**Comment by tpwrules** [comment]
To clarify, a scan grid is essentially a 2D raster where one axis is azimuth and the other is elevation, and the grid value is distance from sensor (instead of e.g. north-south and east-west, plus altitude value on a DEM).

The "mesh scan grid" function connects adjacent points on this raster with triangles. If a particular point is missing, no triangle is created. If the mesh is subsampled then most triangles will be missing as most points will have their adjacent points deleted.

Computing a delauney mesh (perhaps with max triangle angle or distance) on this raster (essentially skipping missing points) BEFORE it is translated from (azimuth, elevation, distance) to (X, Y, Z) would solve my problem and allow me to create an accurate mesh without so many million triangles from a high resolution scan grid.

**Comment by dgirardeau** [comment]
Indeed, I see a difference (as holes will be filled). But there will still be a lot of triangles (even more triangles in fact) since Delaunay triangulation will use all the vertices/points. Or I am missing something else?

**Comment by tpwrules** [comment]
The idea is that you would perform a subsample or some other step to remove most of the points. In this case the regular "mesh scan grid" function won't work since most points would be missing. The delauney triangulation would be able to effectively join the remaining points.
